lib/gui/app/components/source-selector/source-selector.tsx

```diff
@@ -335,7 +336,19 @@ export class SourceSelector extends React.Component<
 				path: selected,
 			});
 		}
-		return new sourceDestination.Http({ url: selected });
+		const splitUrl = selected.split('?');
+		const queryParams = splitUrl.length > 1 && splitUrl[splitUrl.length - 1];
+		const urlSearchParams = !!queryParams
+			? new URLSearchParams(queryParams)
+			: undefined;
+		const params: AxiosRequestConfig | undefined = !!urlSearchParams
+			? Object.fromEntries(urlSearchParams.entries())
+			: undefined;
+		const config = params ?? {};
```

Comment:
This could also be: 

```suggestion
const splitUrl = selected.split('?');
const queryParams = splitUrl.length > 1 && splitUrl[splitUrl.length - 1];
const urlSearchParams = !!queryParams && new URLSearchParams(queryParams);
const params: AxiosRequestConfig | false = !!urlSearchParams && Object.fromEntries(urlSearchParams.entries());
const config = params || {};
```

which is probably shorter, but TBH I prefer in that case to have value or undefined instead of having value or false.
Of course if you prefer the other way around, I'll change it.

Reply:
I prefer to have either the value or undefined too, as the `false` value is still a value and could be misinterpreted in the future